magda-web-client/src/Components/Dataset/Add/ValidationManager.ts

```diff
@@ -51,6 +51,28 @@ export type ElementType =
     | HTMLSpanElement
     | ElementLikeType;
 
+/**
+ * Type for a custom validator. It accepts these parameters:
+ * - fieldValue: The current value of the input
+ * - state: the whole add dataset state data
+ * - validationItem: a ValidationItem, containing information like field label and json path, that might useful for generating a custom error message
+ *
+ * The custom validator may return:
```

Comment:
I reworded this because I had a bit of trouble following the original JSDoc, does this sound right?

If so, I'm a bit uncomfortable with having a single field carrying so much information. Could we split it into something like:

```typescript
{
    valid?: boolean;
    validationMessage?: string;
    useDefaultValidator: boolean;
```

Reply:
Just want to save some typing but I won't be the only one to use this CustomValidator interface --- it might save people some time to look-up the docs.
Updated accordingly~ Please review the document @AlexGilleran